internal/backend/runtime/omni/omni.go

```diff
@@ -245,6 +245,7 @@ func New(talosClientFactory *talos.ClientFactory, dnsService *dns.Service, workl
 		omnictrl.NewTalosExtensionsController(imageFactoryClient),
 		omnictrl.NewTalosUpgradeStatusController(),
 		omnictrl.NewMachineStatusSnapshotController(siderolinkEventsCh),
+		omnictrl.NewLabelsExtractorController[*omni.MachineStatus](),
```

Comment:
please don't forget to mark the new resources as cached in the controller runtime